modules/standard/List.chpl

```diff
@@ -305,12 +323,15 @@ module List {
       _commonInitFromIterable(other);
     }
 
-
     pragma "no doc"
     proc _commonInitFromIterable(iterable) {
       this._firstTimeInitializeArrays();
-      for x in iterable do
-        append(x);
+
+      for x in iterable do {
```

Comment:
@ronawho So after profiling this PR this is trivially the reason comm counts changed in `test/library/packages/Sort/performance/dist-performance.chpl`. The only thing I don't understand is why. Do you think it's worth trying to figure out why this diff caused comm counts to change so drastically? Or should I just revert it and call it a day?

Reply:
Do you remember the motivation for this change? I don't think I understand what's going on here well enough to have an opinion.

Reply:
My rationale for this change was that since this proc is only called in an initializer, I might as well write the loop using internal methods to avoid taking a lock (`list.append` may take a lock).

Reply:
The `pragma "no auto destroy"` exist because the list takes ownership of the copy of the element, so it controls when the copy is destroyed.

I can try `_appendNoLock` but honestly it's probably not worth adding a whole new method, as you say the overhead is not much when uncontested.

I was just curious about whether or not this is worth investigating (as a separate issue or something), as to me it doesn't really seem like the changes should cause such a massive increase in GETs. But maybe it is trivially obvious to you why GETs would increase?

Reply:
I suspect that the pragma code confuses the compile enough that it doesn't remove a copy for the case that the iterator is returning by value. You could check the AST logs around after pass 20  callDestructors for differences to see if that's the case.

Reply:
Awesome! I'll give that a try.

Reply:
No luck, seems more or less the same sans an `initCopy` for the formal tmp in the `append` version vs an `init=` for the variable in the `_appendByRef` version. Iterator was returning refs.